src/main/scala/com/socrata/snapshotter/SodaWatcher.scala

```diff
@@ -52,17 +60,40 @@ class SodaWatcher(curatorFramework: CuratorFramework,
       while(true) {
         Thread.sleep(pauseMS)
         if(!latch.hasLeadership) { return }
-        for(ds <- snapshotDAO.datasetsWithSnapshots().par) {
+        for(resourceName <- snapshotDAO.datasetsWithSnapshots().par) {
           if(!latch.hasLeadership) { return }
           try {
-            for(snapshot <- snapshotDAO.datasetSnapshots(ds)) {
-              if(!latch.hasLeadership) { return }
-              log.info("Purging snapshot {} on dataset {}", snapshot, ds)
-              snapshotDAO.deleteSnapshot(ds, snapshot)
+            val b = new Breaks
+            import b._
+            breakable {
+              for(snapshot <- snapshotDAO.datasetSnapshots(resourceName)) {
+                if(!latch.hasLeadership) { return }
+                log.info("Exporting snapshot {} on dataset {}", snapshot, resourceName.underlying)
+                snapshotDAO.exportSnapshot(resourceName, snapshot) {
+                  case Some(SnapshotDAO.SnapshotInfo(lastModified, stream)) =>
+                    val basename = basenameFor(resourceName, lastModified)
+                    val result =
+                      using(new GZipCompressInputStream(stream, gzipBufferSize)) { inStream =>
+                        blobStoreManager.multipartUpload(inStream, s"$basename.csv.gz")
+                      }
+                    if(result.isRight) {
+                      stream.close()
+                      log.info("Purging snapshot {} on dataset {}", snapshot, resourceName.underlying)
+                      snapshotDAO.deleteSnapshot(resourceName, snapshot)
+                    } else {
+                      log.warn("Problem uploading snapshot {} on dataset {} to amazon; not deleting it")
```

Comment:
At this point, yeah.  That's not customer-facing, but I'll make it less specific.